### DISCUSSION TITLE
Welcome to rutin-tizen Discussions!

### DISCUSSION THREAD

**Discussion by andersondanilo** [andersondanilo/rutin-tizen#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dkson95** [comment]
Dear @andersondanilo

Nice to discuss with you!
I'm Dongkyun Son works for Samsung/ SE team.

How do you think of evaluating *rustin-tizen* on Tizen devices (https://docs.tizen.org/platform/what-is-tizen/overview/) if possible?

Thank you,
Dongkyun.

**Reply by andersondanilo** [reply]
Hello @dkson95
It would be great, however i was running this project only on the tizen wearable (watch) and since the galaxy active went to wear os i believe this project is not so useful anymore. But if there are others devices like CTV who can run native code i believe this project can be adapted

**Reply by dkson95** [reply]
> Hello @dkson95 It would be great, however i was running this project only on the tizen wearable (watch) and since the galaxy active went to wear os i believe this project is not so useful anymore. But if there are others devices like CTV who can run native code i believe this project can be adapted

@andersondanilo I see. Did you have some evaluation report(any regression in binary size, memory footprint, or performance) for the reference ?
Also, now, we have plan to migrate some native(C++) packages into ***RUST*** for the evaluation as POC (targeted to `iot` profile - https://docs.tizen.org/platform/what-is-tizen/profiles/iot/). Would you share some useful ways / steps in migrating to ***RUST*** from existing C/C++ `elm` and `dlog` ?

Thank you!

**Comment by andersondanilo** [comment]
I was doing it in basically to steps, first auto-generating the bindings in [rutin-tizen-sys](https://github.com/andersondanilo/rutin-tizen-sys), and later doing a safe rust interface in this project (rutin-tizen).

I was doing the safe rust interface manually, but today i think that if i could find a way to leverage this process with rust macros it would be much  easier.

Finally i was testing running this example project on the emulator or my galaxy active 2
[rutin-tizen-example](https://github.com/andersondanilo/rutin-tizen-example)

To compile this project we need a lot of env variables and configurations, so i created this cargo plugin to setup it
https://github.com/andersondanilo/cargo-rutin-tizen

so i can simple clone the project, add the tizen studio path in the configuration and build and run it in the emulator
```
cargo tizen dev -e
```

Regarding auto-generating the bindings and compiling, it was based in this [gist](https://gist.github.com/ids1024/00ce4c5e70e90b22866ec7700a00691e) from [@ids1024](https://gist.github.com/ids1024) as mentioned in the rutin-tizen-sys readme
